binaryparse.nim

```diff
@@ -162,40 +166,58 @@ proc replace(node: var NimNode, seenFields: seq[string], parent: NimNode = newId
 
 
 proc decodeType(t: NimNode, stream: NimNode, seenFields: seq[string]):
-  tuple[size: BiggestInt, kind, customReader, customWriter: NimNode] =
+  tuple[size: BiggestInt; endian: Endianness; kind, customReader, customWriter: NimNode] =
   var
     size: BiggestInt
+    endian: Endianness
     kindPrefix: string
     customReader, customWriter: NimNode
   case t.kind:
     of nnkIntLit:
       size = t.intVal
+      endian = system.cpuEndian
       kindPrefix = "int"
     of nnkIdent:
-      case (t.strVal)[0]:
-        of 'u':
-          size = (t.strVal)[1..^1].parseBiggestInt
-          kindPrefix = "uint"
-        of 'f':
-          size = (t.strVal)[1..^1].parseBiggestInt
-          kindPrefix = "float"
-          if size != 32 and size != 64:
-            raise newException(AssertionError,
-              "Only 32 and 64 bit floats are supported")
-        of 's':
-          if (t.strVal).len != 1:
-            size = (t.strVal)[1..^1].parseBiggestInt
-          else:
-            size = 0
-          return (
-            size: BiggestInt(size),
-            kind: newIdentNode("string"),
-            customReader: nil,
-            customWriter: nil
-          )
+      var typ = t.strVal
+
+      if typ.len > 1:
+        case typ[0..1]
+        of "le":
+          endian = littleEndian
+          typ = typ[2..^1]
+        of "be":
+          endian = bigEndian
+          typ = typ[2..^1]
         else:
+          endian = system.cpuEndian
```

Comment:
`endian` is already set to the default value. This branch should just be a discard.